File: components/webxr/android/cardboard_utils.cc
// Copyright 2023 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "base/android/jni_android.h"
#include "components/webxr/android/cardboard_device_provider.h"
#include "device/vr/android/cardboard/cardboard_device_params.h"
#include "device/vr/android/cardboard/mock_cardboard_sdk.h"

// Must come after all headers that specialize FromJniType() / ToJniType().
#include "components/webxr/android/xr_jni_headers/CardboardUtils_jni.h"

namespace webxr {

void JNI_CardboardUtils_NativeUseCardboardV1DeviceParamsForTesting(
    JNIEnv* env) {
  DVLOG(1) << __func__;
  device::CardboardDeviceParams::set_use_cardboard_v1_device_params_for_testing(
      true);
}

void JNI_CardboardUtils_NativeUseCardboardMockForTesting(JNIEnv* env) {
  DVLOG(1) << __func__;
  CardboardDeviceProvider::set_use_cardboard_mock_for_testing(true);
}

jboolean JNI_CardboardUtils_NativeCheckQrCodeScannerWasLaunchedForTesting(
    JNIEnv* env) {
  DVLOG(1) << __func__;
  return device::MockCardboardSdk::
      check_qr_code_scanner_was_launched_for_testing();
}

}  // namespace webxr

File: components/webxr/android/java/src/org/chromium/components/webxr/CardboardUtils.java
// Copyright 2023 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.components.webxr;

import org.jni_zero.JNINamespace;
import org.jni_zero.NativeMethods;

/** Helper class for interacting with Cardboard SDK from Java code. */
@JNINamespace("webxr")
public class CardboardUtils {
    /**
     * Forces to always return Cardboard Viewer v1 device parameters to prevent
     * any disk read or write and the QR code scanner activity to be launched.
     */
    public static void useCardboardV1DeviceParamsForTesting() {
        CardboardUtilsJni.get().nativeUseCardboardV1DeviceParamsForTesting();
    }

    /** Forces to use the mock implementation of the `device::CardboardSdk` interface. */
    public static void useCardboardMockForTesting() {
        CardboardUtilsJni.get().nativeUseCardboardMockForTesting();
    }

    /**
     * Returns true if `MockCardboardSdk::ScanQrCodeAndSaveDeviceParams()` has been executed,
     * otherwise it returns false.
     */
    public static boolean checkQrCodeScannerWasLaunchedForTesting() {
        return CardboardUtilsJni.get().nativeCheckQrCodeScannerWasLaunchedForTesting();
    }

    @NativeMethods
    /* package */ interface CardboardUtilsNative {
        void nativeUseCardboardV1DeviceParamsForTesting();

        void nativeUseCardboardMockForTesting();

        boolean nativeCheckQrCodeScannerWasLaunchedForTesting();
    }
}
